[lens] Support stacking in xy visualization

wylieconlon opened the pull request:
The biggest difference this code makes is to remove `stackSeriesAccessors` from the XY chart state, because we will always stack based on the X accessor if the user chooses to stack their chart. There is no need to duplicate the values in our state.

This won't make sense until we support split series, but sets up the code easily for that: https://github.com/elastic/kibana/issues/38835

<img width="1101" alt="Screenshot 2019-07-08 17 13 10" src="https://user-images.githubusercontent.com/666475/60843094-ad417480-a1a3-11e9-871f-a63800fc3c8c.png">

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/6216/)

wylieconlon commented:
@chrisdavies I agree with you about most of your points, let me respond:

* Stacking only makes sense with split series. I agree, maybe I'll have logic to disable unless there's a split.
* Each of these should be a separate chart type with icon. Sure, this makes sense.
* Stacked lines don't make sense: I'm not sure, stacked area makes sense but I'll ask about stacked lines.
* Checkbox beneath split series: I like the separate icons strategy more.

wylieconlon commented:
@chrisdavies I've implemented in this commit:

* Stacking only allowed with split series
* Separate chart types with icons
* No stacked lines

I also fixed the horizontal charts and added tests to verify that stacking and horizontal charts are working as intended.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/6413/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/6655/)

wylieconlon commented:
Retest

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/6688/)
